### ISSUE TITLE
Global variables assigned to shader uniforms are parsed as relative URLs.

### ISSUE THREAD

**Issue by louh** [tangrams/tangram#449]
**TO REPRODUCE THE ISSUE, FOLLOW THESE STEPS:**

Here is a minimal test case:

```yaml
import:
    - https://tangrams.github.io/blocks/functions/zoom-full.yaml

sources:
    mapzen:
        type: TopoJSON
        url: https://tile.mapzen.com/mapzen/vector/v1/all/{z}/{x}/{y}.topojson

global:
    a: [1, 0, 0]
    b: [0, 0, 1]

layers: 
    earth:
        data: { source: mapzen }
        draw:
            polygons:
                order: 0
                color: grey
    water:
        data: { source: mapzen }
        draw:
            water:
                order: 1
                color: lightblue

styles:
    water:
        base: polygons
        mix: [functions-zoom]
        shaders:
            defines:
                ZOOM_START: 1
                ZOOM_END: 20
            uniforms:
                u_a: global.a
                u_b: global.b
            blocks:
                color: |
                    color.rgb = mix(u_a, u_b,zoom());
```

View in Tangram Play: https://mapzen.com/tangram/play/?scene=https%3A%2F%2Fmapzen.com%2Fapi%2Fscenes%2F7%2F48%2Fresources%2Fscene.yaml#5.022/43.291/-65.717

I am defining colors as `global` variables and then assigning them as uniforms in the shader block at the bottom of the YAML. We expected these values to be passed along to the uniform values. We see instead that Tangram has parsed the references to `global.a` as relative URLs. When we examine the compiled `scene.config` object, we see that `scene.config.styles.water.shaders.uniforms.u_a` now equals `"https://gist.githubusercontent.com/anonymous/804452ab17846fa83732e535d73b7d87/raw/555c9626c3893ea634a7590eaa4e9813df184a95/global.a"` (and similar thing occurs for `u_b`).

Tangram's error in the console is this: 

```
Tangram v0.11.0 [error]: ERROR: 0:758: 'mix' : no matching overloaded function found 
ERROR: 0:758: 'assign' :  cannot convert from 'const float' to 'highp 3-component vector of float'
```

**Comment by blair1618** [comment]
Well, I can verify that this example runs into a genuine bug in tangram-es at least. I would guess that a nearly identical bug is happening in tangram-js but I'll let @bcamper say for sure.

**Comment by bcamper** [comment]
Yeah, there's a lot logic around normalizing the URLs of uniforms/textures according to the scene they were included from. This needs to happen *before* globals are substituted, since you don't have the full set of globals until import/merge process is finished. Probably the easiest way to solve this in JS is to treat `global.` values as reserved for those properties.

**Comment by blair1618** [comment]
Yeah, I'm going to make URL resolution ignore any strings that begin with `global.`, I think we'll just have to live with disallowing `global.jpg` etc. in uniforms.

**Comment by meetar** [comment]
Confirmed locally:

```
> scene.config.styles.water.shaders.uniforms
Object {u_a: "http://localhost:8080/global.a", u_b: "
http://localhost:8080/global.b"}
```

**Comment by blair1618** [comment]
Fix for tangram-es: https://github.com/tangrams/tangram-es/pull/1089

**Comment by bcamper** [comment]
Resolved by https://github.com/tangrams/tangram/commit/b8a1a8d166cde537b5bb09e09c440be5fce9d8c7 and https://github.com/tangrams/tangram/commit/28117bca857e2902d997fd99b9a6a5ac04e98fe7

**Comment by bcamper** [comment]
Released in v0.11.1.
